notebook/static/notebook/js/textcell.js

```diff
@@ -291,6 +291,7 @@ define([
                 // links in markdown cells should open in new tabs
                 html.find("a[href]").not('[href^="#"]').attr("target", "_blank");
                 that.set_rendered(html);
+                that.element.find('div.text_cell_render').find('div').addClass('html_rendered');
```

Comment:
I still think this isn't the right place to do it, and the class is rendered_html, not html_rendered.  Look at where the html output adds the class (append_html), and there is a similar place in the markdown output (append_markdown) where it can be added, without making any jquery calls.